packages/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -40,62 +40,136 @@ interface Queues {
+		this._queues.verified.enqueueMany(block.transactions);
+	}
 
-	private expireTransactions(): void {}
+	public onNewBlock(block: Block): void {
+		// Remove transactions in the transaction pool which were included in the new block
+		this.removeTransactionsFromQueues(
+			this._queues,
+			queueCheckers.checkTransactionForId(block.transactions),
+		);
+
+		const { received, validated, ...otherQueues } = this._queues;
+		// Remove transactions from the verified, pending and ready queues which were sent from the accounts in the new block
+		const transactionsFromAffectedAccounts = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionForSenderPublicKey(block.transactions),
+		);
+
+		// Remove all transactions from the verified, pending and ready queues if they are of a type which includes unique data and that type is included in the block
+		// TODO: remove the condition for checking `containsUniqueData` exists, because it should always exist
+		const blockTransactionsWithUniqueData = block.transactions.filter(
+			(transaction: Transaction) =>
+				transaction.containsUniqueData && transaction.containsUniqueData(),
+		);
+		const transactionsOfTypesWithUniqueData = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionForTypes(blockTransactionsWithUniqueData),
+		);
+
+		// Add transactions which need to be reverified to the validated queue
+		this._queues.validated.enqueueMany([
+			...transactionsFromAffectedAccounts,
+			...transactionsOfTypesWithUniqueData,
+		]);
+	}
 
-	private processVerifiedTransactions(): void {}
+	public onRoundRollback(delegates: ReadonlyArray<string>): void {
+		// Move transactions from the verified, pending and ready queues to the validated queue which were sent from delegate accounts
+		const { received, validated, ...otherQueues } = this._queues;
+		const senderProperty: queueCheckers.TransactionFilterableKeys =
+			'senderPublicKey';
+		const transactionsFromAffectedAccounts = this.removeTransactionsFromQueues(
+			otherQueues,
+			queueCheckers.checkTransactionPropertyForValues(
+				delegates,
+				senderProperty,
+			),
+		);
+
+		this._queues.validated.enqueueMany(transactionsFromAffectedAccounts);
+	}
 
-	private validateReceivedTransactions(): void {}
+	public validateTransactionAgainstTransactionsInPool(
+		transaction: Transaction,
+	): boolean {
+		// TODO: remove the condition for checking `verifyTransactionAgainstOtherTransactions` exists, because it should always exist
+		return transaction.verifyTransactionAgainstOtherTransactions
+			? transaction.verifyTransactionAgainstOtherTransactions([
+					...this.queues.ready.transactions,
+					...this.queues.pending.transactions,
+					...this.queues.verified.transactions,
+			  ])
+			: true;
+	}
 
-	private verifyValidatedTransactions(
-		transactions: ReadonlyArray<Transaction>,
-	): void {}
+	private removeTransactionsFromQueues(
+		queues: Queues,
```

Comment:
Is it confusing because i am internally using this._queues? I can see that I don't need to do that, and I can just do: `.map(queueName => queues[queueName].removeFor(condition))`. Passing in queueName as input would just make the function calls a bit dirty, I think.